tutorials/untyped/untyped.md

```diff
@@ -59,10 +59,10 @@ shows how to print out the initial untyped capabilities provided from `seL4_Boot
 ### Retyping
 
 Untyped capabilities have a single invocation: [seL4_Untyped_Retype](https://docs.sel4.systems/ApiDoc.html#retype)
-which is used to create a new capability from an untyped. Specifically, the new capability created by
-a retype invocation provides access to the a subset of the memory range granted by the original untyped,
- with a specific type. New capabilities created by retyping an untyped object are referred to as *children* of
- that untyped object.
+which is used to create a new capability to a new object. The new object is a subset of the memory range granted
+by the original untyped and has a specific type. The new capability allows access to this object and is stored
+in an empty slot in an existing CNode. New capabilities created by retyping an untyped object are referred to as
+*children* of that untyped object.
```

Comment:
This is the main part that is problematic. The thing the previous text was trying to avoid to talk about was that there are different kinds of capabilities:
- caps that do not point to memory objects (like `IRQControl`)
- caps that *do* point  to memory objects
- untyped caps, that are responsible for memory ranges and allocation from those

Maybe we should just be explicit about that. The `retype` operation deals with the latter two types of capabilities. For object capabilities it creates new objects and new caps to them. For untyped capabilities, it creates new sub-untypeds that intended for delegating authority to parts of the memory region and allocating from there independently.

Reply:
Yes, I think being explicit about it either creating an object or dividing up the untyped range is a good idea.

I think the most confusing part of the original text is "...which is used to create a new capability from an untyped". To me the important thing that retype does is creating the object, not the capability, except as you pointed out when making sub-untypeds.